src/platform/Linux/PlatformManagerImpl.cpp

```diff
@@ -175,12 +175,15 @@ CHIP_ERROR PlatformManagerImpl::_InitChipStack()
 
     // Initialize the configuration system.
     ReturnErrorOnFailure(Internal::PosixConfig::Init());
-    SetDeviceInstanceInfoProvider(&DeviceInstanceInfoProviderMgrImpl());
 
     // Call _InitChipStack() on the generic implementation base class
     // to finish the initialization process.
     ReturnErrorOnFailure(Internal::GenericPlatformManagerImpl_POSIX<PlatformManagerImpl>::_InitChipStack());
 
+    // Now set up our device instance info provider.  We couldn't do that
+    // earlier, because the generic implementation sets a generic one.
```

Comment:
The problem is the "generic implementation" setting a "generic one" (which cannot properly be right). We should move the device instance info provider to the Server and have a getter for it.

Reply:
I absolutely agree.  I'm just trying to unblock https://github.com/project-chip/connectedhomeip/issues/21139 without rewriting the whole thing in the process because I don't have time for that right this second and because I am not quite sure what shape the rewrite would have....

Reply:
Ideally, this stuff would not use configuration manager at all, by the way.